### PR TITLE
Fix bug in indexer chunking recursion

### LINKED ISSUE #348: Bug in indexer chunking recursion
There's a bug in the recursion for when the indexer is chunking out blocks to validate that is causing a 413 error in the tests.

### Steps to reproduce

* Ethereum_jsonrpc test "transparently splits batch payloads that would trigger a 413 Request Entity Too Large" fails

### PR DESCRIPTION
**Pull request by Stamates**
Resolves #348 

## Motivation

Fix failing test

### Bug Fixes
* Added missing function pattern match to handle the final step in the chunking recursion where a non-nested list is sent into the chunking function.

### CODE REVIEW

File: apps/ethereum_jsonrpc/lib/ethereum_jsonrpc.ex [hunk 1]
```diff
@@ -280,6 +280,10 @@ defmodule EthereumJSONRPC do
     end
   end
 
+  defp chunked_json_rpc(url, tail, options, decoded_response_bodies) do
```
Comment (amandasposito) [inline apps/ethereum_jsonrpc/lib/ethereum_jsonrpc.ex]: WDYT of a test for this?

Comment (Stamates) [reply apps/ethereum_jsonrpc/lib/ethereum_jsonrpc.ex]: In this case, the code was added to pass an existing test (so it is tested), but also, standard practice would be to not write tests for private functions.

Comment (amandasposito) [reply apps/ethereum_jsonrpc/lib/ethereum_jsonrpc.ex]: I meant a test that uses this private function, so we can have a better context of how this solves the problem.

If it is an intermittent test in the suit, an output of the failing test in the PR description should do the work. 👍

Comment (Stamates) [reply apps/ethereum_jsonrpc/lib/ethereum_jsonrpc.ex]: Upon further review, the original code was fine (just a little confusing in terms of how it worked).  I must have had some weird inconsistent error that was causing the test failure on Friday when I worked on this.  Thanks for reviewing, I'll close this PR.

**Pull request closed without merging**